crates/uv-configuration/src/dev.rs

```diff
@@ -1,445 +1,350 @@
-    pub fn only(&self) -> bool {
-        matches!(self, Self::Only { .. })
-    }
-
-    /// Returns the option that was used to request the groups, if any.
-    pub fn as_flag(&self) -> Option<Cow<'_, str>> {
-        match self {
-            Self::Include { include, exclude } => match include {
-                IncludeGroups::All => Some(Cow::Borrowed("--all-groups")),
-                IncludeGroups::Some(groups) => match groups.as_slice() {
-                    [] => match exclude.as_slice() {
-                        [] => None,
-                        [group] => Some(Cow::Owned(format!("--no-group {group}"))),
-                        [..] => Some(Cow::Borrowed("--no-group")),
-                    },
-                    [group] => Some(Cow::Owned(format!("--group {group}"))),
-                    [..] => Some(Cow::Borrowed("--group")),
-                },
-            },
-            Self::Only { include, exclude } => match include.as_slice() {
-                [] => match exclude.as_slice() {
-                    [] => None,
-                    [group] => Some(Cow::Owned(format!("--no-group {group}"))),
-                    [..] => Some(Cow::Borrowed("--no-group")),
-                },
-                [group] => Some(Cow::Owned(format!("--only-group {group}"))),
-                [..] => Some(Cow::Borrowed("--only-group")),
-            },
-            Self::Explicit { include } => match include.as_slice() {
-                [] => Some(Cow::Borrowed("--no-default-groups")),
-                [group] => Some(Cow::Owned(format!("--group {group}"))),
-                [..] => Some(Cow::Borrowed("--group")),
-            },
-        }
-    }
-
-    /// Iterate over all groups referenced in the [`GroupsSpecification`].
-    pub fn names(&self) -> impl Iterator<Item = &GroupName> {
-        match self {
-            GroupsSpecification::Include { include, exclude } => {
-                Either::Left(include.names().chain(exclude.iter()))
-            }
-            GroupsSpecification::Only { include, exclude } => {
-                Either::Left(include.iter().chain(exclude.iter()))
-            }
-            GroupsSpecification::Explicit { include } => Either::Right(include.iter()),
-        }
-    }
-
-    /// Returns `true` if the specification includes the given group.
-    pub fn contains(&self, group: &GroupName) -> bool {
-        match self {
-            GroupsSpecification::Include { include, exclude } => {
-                // For `--all-groups`, the group is included unless it is explicitly excluded.
-                include.contains(group) && !exclude.contains(group)
+        let include = if all_groups {
+            // If this is set we can ignore group/only_group/defaults as irrelevant
+            IncludeGroups::All
+        } else {
+            // Merge all these lists, they're equivalent now
+            group.append(&mut only_group);
+            // Throw out the defaults if those are disabled
```

Comment:
Perhaps something more aligned with the action we're taking
```suggestion
            // Include the defaults if they're not disabled
```